pkg/resource/deployment.go

```diff
@@ -104,11 +110,17 @@ func (gd *generatorDeployment) expected() (runtime.Object, error) {
 			},
 			Template: podTemplateSpec,
 			Strategy: appsapi.DeploymentStrategy{
-				Type: appsapi.DeploymentStrategyType(strategy),
+				Type: appsapi.DeploymentStrategyType(deployStrategy),
 			},
 		},
 	}
 
+	dgst, err := strategy.Checksum(deploy)
+	if err != nil {
+		return nil, err
+	}
+	deploy.ObjectMeta.Annotations[parameters.ChecksumOperatorAnnotation] = dgst
```

Comment:
Do you need this only for backward compatibility?

Reply:
No, we need this because `resourceapply.ApplyDeployment()`  takes into account only `ObjectMeta` when [detecting changes](https://github.com/openshift/library-go/blob/master/pkg/operator/resource/resourceapply/apps.go#L36).